packages/media-query-mixin/test/media-query-mixin.test.ts

```diff
@@ -0,0 +1,156 @@
+import { LitElement, html, customElement } from 'lit-element';
+import { fixture } from '@open-wc/testing-helpers';
+import { MediaQueryMixin, mediaProperty } from '../media-query-mixin';
+
+const { expect } = chai;
+
+const listeners: { [key: string]: () => void } = {};
+const queryMatches: { [key: string]: boolean } = {};
+
+const runQuery = (query: string, matches: boolean) => {
+  queryMatches[query] = matches;
+  listeners[query] && listeners[query]();
+};
+
+const FULLSCREEN = '(max-width: 420px)';
+const RESPONSIVE = '(max-width: 600px)';
+
+const origMatchMedia = window.matchMedia;
+window.matchMedia = (query: string) => {
+  return {
+    get matches() {
+      return queryMatches[query] as boolean;
+    },
+    addListener(listener: () => void) {
+      listeners[query] = listener;
+    },
+    removeListener(_listener) {
+      delete listeners[query];
+    }
+  } as MediaQueryList;
+};
+
+class MqElementBase extends MediaQueryMixin(LitElement) {
+  render() {
+    return html`
+      <div part="content">Content</div>
+    `;
+  }
+}
+
+describe('MediaQueryMixin', () => {
+  describe('mediaProperty decorator', () => {
+    @customElement('mq-basic-element')
+    class MqBasicElement extends MqElementBase {
+      @mediaProperty({ media: FULLSCREEN }) fullscreen: boolean | null | undefined;
+    }
+
+    let element: MqBasicElement;
+
+    before(() => {
+      runQuery(FULLSCREEN, true);
+    });
+
+    beforeEach(async () => {
+      element = await fixture(`<mq-basic-element></mq-basic-element>`);
+    });
+
+    it('should set default property value based on media query', () => {
+      expect(element.fullscreen).to.be.true;
+    });
+
+    it('should toggle property value when query does not match', async () => {
+      runQuery(FULLSCREEN, false);
+      await element.updateComplete;
+      expect(element.fullscreen).to.be.false;
+    });
+
+    it('should disallow direct property modification attempt', async () => {
```

Comment:
Should we consider special case of unsetting the prop, to clean up the media query checks by setting prop to `null`
and add test for it?

Reply:
Discussed in person that we want to align it with how `iron-media-query` works.